### ISSUE TITLE
cocsistent-return reports error in return inside ES6 class constructor

### ISSUE THREAD

**Issue by damonmaria** [eslint/eslint#7790]
**Tell us about your environment**

ESLint: 3.12.2
Node: v4.6.2
yarn: 0.17.10

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel-ESLint 7.1.1

**Please show your full configuration:**

```
parser: babel-eslint
rules:
  consistent-return: error
```

**What did you do? Please include the actual source code causing the issue.**

```js
class A {
  constructor(b) {
    if (b) {
      return b;
    }
  }
}
```

**What did you expect to happen?**

The return shouldn't be a problem, as it's in a constructor. 

ES5 constructors do not cause a problem:
```js
function F(b) {
  if (b) {
    return b;
  }
}
```

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint reports consistent-return

**Comment by ndaidong** [comment]
cocsistent-return here means that you need to define which would be returned when b is Falsy.

I guest something like below will not throw any error:

```
class A {
  constructor(b) {
    if (b) {
      return b;
    }
    return DEFAULT_VALUE;
  }
}
```

**Comment by damonmaria** [comment]
Correct. That passes. But it should match how it works for ES5 constructors and also what the docs imply:

> This rule ignores function definitions where the name begins with an uppercase letter, because constructors (when invoked with the new operator) return the instantiated object implicitly if they do not return another object explicitly.

I just don't think it's been updated handle ES6 constructors in the same way.

**Comment by not-an-aardvark** [comment]
Thanks for the report. It looks like you're correct -- if the rule ignores functions that start with a capital letter because they're constructors, then it should probably ignore constructors in class definitions too.

Would you like to submit a pull request to fix it? (It's no problem if you're not able to, but we try to give people a chance to contribute.)

**Comment by damonmaria** [comment]
I would love to say yes but to be honest I just don't have the time. Apologies :(

**Comment by not-an-aardvark** [comment]
No worries, thanks again for reporting the issue!
